src/Microsoft.DotNet.Watcher.Core/Internal/FileWatcher/PollingFileWatcher.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    internal class PollingFileWatcher : IFileSystemWatcher
+    {
+        // The minimum interval to rerun the scan
+        private static readonly TimeSpan _minRunInternal = TimeSpan.FromSeconds(.5);
+
+        private readonly DirectoryInfo _watchedDirectory;
+
+        private Dictionary<string, FileMeta> _knownEntities = new Dictionary<string, FileMeta>();
+        private Dictionary<string, FileMeta> _tempDictionary = new Dictionary<string, FileMeta>();
+        private HashSet<string> _changes = new HashSet<string>();
+
+        private Thread _pollingThread;
+        private bool _raiseEvents;
+
+        private bool _disposed;
+
+        public PollingFileWatcher(string watchedDirectory)
+        {
+            if (string.IsNullOrEmpty(watchedDirectory))
+            {
+                throw new ArgumentNullException(nameof(watchedDirectory));
+            }
+
+            _watchedDirectory = new DirectoryInfo(watchedDirectory);
+
+            _pollingThread = new Thread(new ThreadStart(PollingLoop));
+            _pollingThread.IsBackground = true;
+            _pollingThread.Name = nameof(PollingFileWatcher);
+            _pollingThread.Start();
+        }
+
+        public event EventHandler<string> OnFileChange;
+
+#pragma warning disable CS0067 // not used
```

Comment:
More info why this is disabled.  I.e. we need it because of the interface, but compiler requires all the events to be used. Or something along those lines :)